### ISSUE TITLE
Format on save is extremely slow [20+ sec]

### ISSUE THREAD

**Issue by alechko** [valeryan/vscode-phpsab#35]
Hello!

I have an issue with phpsab, every save it takes ~5 seconds to start the phpsab and another 10-15 to complete the task, which is super slow. During that time the file remains unsaved and you cannot see the changes you've made. 

Edit: If I use phpcbf from the cli, it takes 49ms to complete the same file.

My vscode settings: 
```
{
  "[php]": {
    "editor.tabSize": 2,
    "editor.defaultFormatter": "valeryanm.vscode-phpsab",
    "editor.formatOnSave": true
  },
  "phpcs.standard": "Drupal",
  "phpsab.fixerEnable": true,
  "phpsab.snifferEnable": false,
  "phpsab.standard": "Drupal",
  "phpsab.debug": false,
  "phpsab.snifferShowSources": true
}
```

vscode: 1.50
phpsab: v0.0.11
os: osx 10.15.7

![Kapture 2020-10-28 at 9 42 31](https://user-images.githubusercontent.com/1391846/97406571-181b3f80-1902-11eb-84ea-3e5398dc99b2.gif)

**Comment by valeryan** [comment]
I use a mac as my primary driver and I can't replicate your issue. Try running phpcbf directly from the command line against this file with the same ruleset. This could be an issue with phpcbf itself. It could be an issue with the drupal ruleset.

**Comment by alechko** [comment]
@valeryan I've already mentioned that running phpcbf from cli takes 49ms for the same file with the same standard.

**Comment by valeryan** [comment]
@alechko I read your issue from my email and it did not include your edit. 

The best I can do without being able to replicate this is walk you through some of the things I would try if this was happening to me. 

1. Check your activity monitor for any hung processes. Under Applications>Utilities>Activity Monitor
2. Try a different ruleset
3. Try a different file
4. Disable all extension and then enable phpsab to see if problem persist. If it has gone away start enabling extension until the problem returns.

**Comment by alechko** [comment]
I've taken your advice, created a new php file and did some experiments. I've managed to narrow down the issue to a commenting issue.

I've created 2 simple files with one function and lots of commented text, the issue seems to come from a comment that cannot be formatted and having lint problem.

[here is the gist](https://gist.github.com/alechko/394a97649460929a5347d8149f4e1f86) of the 2 files, one is test-normal.php and the other test-slow.php

The difference between them is the first comment removed from the normal file.

It seems that in that specific case phpcbf is the culprit, as running it from cli results in ~15sec execution time, but on other files I've worked on phpcbf executes super fast, while phpsab takes long time to complete, so I'm not sure that my example is perfect, but I guess it's phpcbf problem indeed, and this probably needs to be redirected to phpcbf.

Thanks!
